src/control/pbin/stdio_test.go

```diff
@@ -0,0 +1,140 @@
+//
+// (C) Copyright 2019 Intel Corporation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// GOVERNMENT LICENSE RIGHTS-OPEN SOURCE SOFTWARE
+// The Government's rights to use, modify, reproduce, release, perform, display,
+// or disclose this software are subject to the terms of the Apache License as
+// provided in Contract No. 8F-30005.
+// Any reproduction of computer software, computer software documentation, or
+// portions thereof marked with this legend must also reproduce the markings.
+//
+package pbin_test
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"os"
+	"os/exec"
+	"testing"
+
+	"github.com/daos-stack/daos/src/control/pbin"
+)
+
+const (
+	childModeEnvVar = "GO_TESTING_CHILD_MODE"
+	childModeEcho   = "MODE_ECHO"
+	childModeReqRes = "MODE_REQ_RES"
+	testMsg         = "hello world"
+)
+
+func childErrExit(err error) {
+	if err == nil {
+		err = errors.New("unknown error")
+	}
+	fmt.Fprintf(os.Stderr, "CHILD ERROR: %s\n", err)
+	os.Exit(1)
+}
+
+func echo() {
+	sc := pbin.NewStdioConn("child", "parent", os.Stdin, os.Stdout)
+	lis := pbin.NewStdioListener(sc)
+
+	conn, err := lis.Accept()
+	if err != nil {
+		childErrExit(err)
+	}
+	defer lis.Close()
+
+	buf := make([]byte, len(testMsg))
+	var total int
+	for {
+		read, err := conn.Read(buf)
+		if err != nil {
+			childErrExit(err)
+		}
+		total += read
+		if total == len(buf) {
+			_, err = conn.Write(buf)
+			if err != nil {
+				childErrExit(err)
+			}
+			break
+		}
+	}
+}
+
+func TestMain(m *testing.M) {
+	switch os.Getenv(childModeEnvVar) {
+	case "":
+		os.Exit(m.Run())
+	case childModeEcho:
+		echo()
+	case childModeReqRes:
+		reqRes()
```

Comment:
Can only have one TestMain() function per package, and it was already in this file to start with. It would be a little odd to move that test helper function which is only used in the other file into this one, IMO. I can add a comment here if that makes things clearer, though.

Reply:
Would it be clearer if I added a pbin_test.go to hold the TestMain() function and some comments indicating that the child functions are defined in their respective test files?

Reply:
Yeah, I think that would clarify the intent.

Reply:
Done.